server/backend/backend.go

```diff
@@ -646,19 +646,73 @@ func (b *Backend) DisableAutoMine() {
 	b.automine = false
 }
 
-func (b *Backend) GetTransactionResultByIndex(context.Context, flowgo.Identifier, uint32) (*access.TransactionResult, error) {
-	// TODO: implement
-	panic("GetTransactionResultByIndex not implemented")
+func (b *Backend) GetTransactionResultByIndex(ctx context.Context, id sdk.Identifier, index uint32) (*sdk.TransactionResult, error) {
+	results, err := b.GetTransactionResultsByBlockID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if uint32(len(results)) <= index {
+		return nil, status.Error(codes.NotFound, "TransactionResult not found")
+	}
+	return results[index], nil
 }
 
-func (b *Backend) GetTransactionsByBlockID(ctx context.Context, id flowgo.Identifier) ([]*flowgo.TransactionBody, error) {
-	// TODO: implement
-	panic("GetTransactionsByBlockID not implemented")
+func (b *Backend) GetTransactionsByBlockID(ctx context.Context, id sdk.Identifier) (result []*sdk.Transaction, err error) {
```

Comment:
could you add this to the backend_test?

Reply:
yeah actually I want to also test GRPC, but somehow flow-go-sdk I think does not support yet this. Is there any tool I can use?

Reply:
I think just adding to those tests using the existing mock is fine. For GRPC we should probably write new e2e tests which are currently non-existing. So I feel that might be another PR just addressing that.

Reply:
@sideninja I need a bit help here, tried this test but I stuck I guess.